core/os/android/adb/screen.go

```diff
@@ -105,18 +105,17 @@ func (b *binding) UnlockScreen(ctx context.Context) (bool, error) {
 		// Devices may do unexpected transitions between screen
 		// states, so this code does not try to be smart about
 		// expected state changes: unless screenOnUnlocked, apply the
-		// wakeup (put screen on) and menu (unlock screen if no
-		// credentials required) key event sequence.
+		// wakeup (put screen on) and dismiss-keyguard (unlock screen
+		// if no credentials required) sequence.
 		if err := b.KeyEvent(ctx, android.KeyCode_Wakeup); err != nil {
 			return false, err
 		}
-		if err := b.KeyEvent(ctx, android.KeyCode_Menu); err != nil {
+		if err := b.Shell("wm", "dismiss-keyguard").Run(ctx); err != nil {
```

Comment:
What version of Android introduced `wm dismiss-keyguard`? Was it added before Q?

Reply:
It seems to have been around for quite some time. This refactor commit from 2017 moved it to a new file within Android:
https://cs.android.com/android/_/android/platform/frameworks/base/+/5c3296af2d2b9717065b632c1848a45b514540a2

So it predates 2017, I'm confident it was there by Q.